packages/eas-cli/src/submissions/android/AndroidSubmitCommand.ts

```diff
@@ -0,0 +1,233 @@
+import { Result, result } from '@expo/results';
+import validator from 'validator';
+
+import log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import { ensureLoggedInAsync } from '../../user/actions';
+import {
+  ArchiveFileSource,
+  ArchiveFileSourceType,
+  ArchiveSource,
+  ArchiveTypeSource,
+  ArchiveTypeSourceType,
+} from '../archive-source';
+import { SubmissionPlatform } from '../types';
+import { getExpoConfig } from '../utils/config';
+import { AndroidPackageSource, AndroidPackageSourceType } from './AndroidPackageSource';
+import { AndroidArchiveType, ReleaseStatus, ReleaseTrack } from './AndroidSubmissionConfig';
+import AndroidSubmitter, { AndroidSubmissionOptions } from './AndroidSubmitter';
+import { ServiceAccountSource, ServiceAccountSourceType } from './ServiceAccountSource';
+import { AndroidSubmissionContext, AndroidSubmitCommandFlags } from './types';
+
+class AndroidSubmitCommand {
+  static createContext(
+    projectDir: string,
+    commandFlags: AndroidSubmitCommandFlags
+  ): AndroidSubmissionContext {
+    return {
+      projectDir,
+      commandFlags,
+    };
+  }
+
+  constructor(private ctx: AndroidSubmissionContext) {}
+
+  async runAsync(): Promise<void> {
+    const projectId = await this.getProjectIdAsync();
+    log.addNewLineIfNone();
+
+    const submissionOptions = this.getAndroidSubmissionOptions(projectId);
+    const submitter = new AndroidSubmitter(this.ctx, submissionOptions);
+    await submitter.submitAsync();
+  }
+
+  private getAndroidSubmissionOptions(projectId: string): AndroidSubmissionOptions {
+    const androidPackageSource = this.resolveAndroidPackageSource();
+    const track = this.resolveTrack();
+    const releaseStatus = this.resolveReleaseStatus();
+    const archiveSource = this.resolveArchiveSource(projectId);
+    const serviceAccountSource = this.resolveServiceAccountSource();
+
+    const errored = [
+      androidPackageSource,
+      track,
+      releaseStatus,
+      archiveSource,
+      serviceAccountSource,
+    ].filter(r => !r.ok);
+    if (errored.length > 0) {
+      const message = errored.map(err => err.reason?.message).join('\n');
+      log.error(message);
+      throw new Error('Failed to submit the app');
+    }
+
+    return {
+      projectId,
+      androidPackageSource: androidPackageSource.enforceValue(),
+      track: track.enforceValue(),
+      releaseStatus: releaseStatus.enforceValue(),
+      archiveSource: archiveSource.enforceValue(),
+      serviceAccountSource: serviceAccountSource.enforceValue(),
+    };
+  }
+
+  private async getProjectIdAsync(): Promise<string> {
+    const user = await ensureLoggedInAsync();
```

Comment:
Maybe pre-authenticate the user in the `run` function - https://github.com/expo/eas-cli/pull/30/files#diff-f916f1e07c79cf8c4c17df1404389994227369f20d0f3ce47280aef8b5cf4965R70 - to be sure that if the user is not logged in he cannot use this command.